core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -365,7 +350,7 @@ private void startNewPage()
             currentPageBuilder = currentPageBuilder.newPageBuilderLike();
         }
         else {
-            currentPageBuilder = new PageBuilder(types);
+            currentPageBuilder = new PageBuilder(OUTPUT_PAGE_SIZE, Integer.MAX_VALUE, types);
```

Comment:
Why? Why not to start with minimal builder for first page so that we don't have to allocate for 16k positions and waste memory?

Reply:
It doesn't seem to have impact on perf. I reverted it.
```
pre size to OUTPUT_PAGE_SIZE
Benchmark                                   (channelCount)  (dataType)  (groupCount)  (hashEnabled)  Mode  Cnt    Score   Error  Units
BenchmarkGroupByHash.groupByHashPreCompute               2      BIGINT          8000           true  avgt   20   33.467 ± 2.076  ns/op
BenchmarkGroupByHash.groupByHashPreCompute               2      BIGINT       3000000           true  avgt   20  234.790 ± 2.115  ns/op

no pre size
Benchmark                                   (channelCount)  (dataType)  (groupCount)  (hashEnabled)  Mode  Cnt    Score   Error  Units
BenchmarkGroupByHash.groupByHashPreCompute               2      BIGINT          8000           true  avgt   20   34.916 ± 1.100  ns/op
BenchmarkGroupByHash.groupByHashPreCompute               2      BIGINT       3000000           true  avgt   20  233.863 ± 6.016  ns/op
```